src/SprykerEco/Zed/AmazonQuicksight/Business/Processor/AssetBundleQuicksightUserProcessor.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\AmazonQuicksight\Business\Processor;
+
+use Generated\Shared\Transfer\QuicksightUserTransfer;
+use Generated\Shared\Transfer\UserCollectionResponseTransfer;
+use Generated\Shared\Transfer\UserTransfer;
+use SprykerEco\Zed\AmazonQuicksight\Business\Creator\QuicksightUserCreatorInterface;
+use SprykerEco\Zed\AmazonQuicksight\Business\Updater\QuicksightUserUpdaterInterface;
+
+class AssetBundleQuicksightUserProcessor implements AssetBundleQuicksightUserProcessorInterface
+{
+    /**
+     * @uses \SprykerEco\Zed\AmazonQuicksight\AmazonQuicksightConfig::QUICKSIGHT_USER_ROLE_AUTHOR
+     *
+     * @var string
+     */
+    protected const QUICKSIGHT_USER_ROLE_AUTHOR = 'AUTHOR';
+
+    /**
+     * @uses \SprykerEco\Zed\AmazonQuicksight\AmazonQuicksightConfig::QUICKSIGHT_USER_ROLE_READER
+     *
+     * @var string
+     */
+    protected const QUICKSIGHT_USER_ROLE_READER = 'READER';
+
+    /**
+     * @var \SprykerEco\Zed\AmazonQuicksight\Business\Creator\QuicksightUserCreatorInterface
+     */
+    protected QuicksightUserCreatorInterface $quicksightUserCreator;
+
+    /**
+     * @var \SprykerEco\Zed\AmazonQuicksight\Business\Updater\QuicksightUserUpdaterInterface
+     */
+    protected QuicksightUserUpdaterInterface $quicksightUserUpdater;
+
+    /**
+     * @param \SprykerEco\Zed\AmazonQuicksight\Business\Creator\QuicksightUserCreatorInterface $quicksightUserCreator
+     * @param \SprykerEco\Zed\AmazonQuicksight\Business\Updater\QuicksightUserUpdaterInterface $quicksightUserUpdater
+     */
+    public function __construct(
+        QuicksightUserCreatorInterface $quicksightUserCreator,
+        QuicksightUserUpdaterInterface $quicksightUserUpdater
+    ) {
+        $this->quicksightUserCreator = $quicksightUserCreator;
+        $this->quicksightUserUpdater = $quicksightUserUpdater;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UserTransfer $userTransfer
+     *
+     * @return \Generated\Shared\Transfer\UserCollectionResponseTransfer
+     */
+    public function processQuicksightUserBeforeAnalyticsEnabling(
+        UserTransfer $userTransfer
+    ): UserCollectionResponseTransfer {
+        $quicksightUserTransfer = $userTransfer->getQuicksightUser();
```

Comment:
Maybe we need to do a DB call to find out if the current user has a quicksight user instead of relying on the provided data, what do you think?